agent/engine/docker_task_engine.go

```diff
@@ -238,9 +238,19 @@ func (engine *DockerTaskEngine) synchronizeState() {
 
 	tasks := engine.state.AllTasks()
 	for _, task := range tasks {
+		// Update the registry credentials of container if needed
+		// The following situations should be taken care of:
+		// 1. Agent restart:
+		//     * task is not started, task needs to wait for credentials to pull
+		//     * task already in running state, no need to wait
+		// 2. ACS send new payload message update the task
 		conts, ok := engine.state.ContainerMapByArn(task.Arn)
 		if !ok {
-			engine.startTask(task)
+			if task.ShouldWaitForExecutionCredentials() {
+				seelog.Info("engine synchronize: task waiting for execution role: %s", task.String())
```

Comment:
Log message should begin with upper case letter. Also, `"Synchronizing state: ..."` might be a better message